pkg/buses/monitor.go

```diff
@@ -192,9 +203,16 @@ func (h MonitorEventHandlerFuncs) onSubscribe(subscription *channelsv1alpha1.Sub
 			cond = util.NewSubscriptionCondition(channelsv1alpha1.SubscriptionDispatching, corev1.ConditionTrue, successSynced, "Subscription dispatcher successfully created")
 		}
 		util.SetSubscriptionCondition(&subscriptionCopy.Status, *cond)
-		_, errS := monitor.clientset.ChannelsV1alpha1().Subscriptions(subscription.Namespace).Update(subscriptionCopy)
-		if errS != nil {
-			glog.Warningf("Could not update status: %v", errS)
+		if !equality.Semantic.DeepEqual(subscription.Status, subscriptionCopy.Status) {
+			// status has changed, update subscription
+			_, errS := monitor.eventingClient.ChannelsV1alpha1().Subscriptions(subscription.Namespace).Update(subscriptionCopy)
```

Comment:
It feels strange to be tracking two `error`s here.

Reply:
it is a bit funky. We want always want to update the resource status, if the original err being recorded in the resource status, we want to return that error since it's more meaningful. If the update failed, we still want to return that error to trigger the work queue to process the resource again as the update can fail because of a failed optimistic lock.

Will add a comment in the code